docs: improve single root element note about comments (#4245)
nuxt/framework: docs/content/2.guide/3.directory-structure/10.pages.md

</template>
```

> Note that pages **must have a single root element** to allow route transitions between pages, and that HTML comments are considered elements as well.
> This means that when the route is server rendered, or statically generated, you will be able to see its contents correctly, but when you navigate towards that route during client side navigation the transition between routes will fail and you'll see that the route will not be rendered.
> See the code examples below for pragmatic examples of what constitutes a correctly defined page and what does not.

Reviewer: ```suggestion
Pages **must have a single root element** to allow route transitions between pages. (HTML comments are considered elements as well.)

This means that when the route is server rendered, or statically generated, you will be able to see its contents correctly, but when you navigate towards that route during client side navigation the transition between routes will fail and you'll see that the route will not be rendered.

Here are some examples to illustrate what a page with a single root element looks like:
```

::code-group

> ```vue [pages/working-route.vue]

Reviewer: ```suggestion
```vue [pages/working.vue]
```

<template>
 <div>
  <!-- This page correctly has only one single root element -->
  Page content
 </div>
</template>
```

> ```vue [pages/bad-route-example-1.vue]
> <template>
>   <!-- This page will not render when route changes during client side navigation, because of this comment -->
>   <div> Page content </div>
> </template>
> ```
> 
> ```vue [pages/bad-route-example-2.vue]

Reviewer: ```suggestion
```vue [pages/bad-1.vue]
<template>
  <!-- This page will not render when route changes during client side navigation, because of this comment -->
  <div>Page content</div>
</template>
```

```vue [pages/bad-2.vue]
```

<template>

>  <div> This page </div>
>  <div> Has more than one root element </div>
>  <div> And will not render when route changes during client side navigation </div>

Reviewer: ```suggestion
 <div>This page</div>
 <div>Has more than one root element</div>
 <div>And will not render when route changes during client side navigation</div>
```

::alert{type=warning}

> Note that pages **must have a single root element** to allow route transitions between pages.
> Also note that HTML comments blocks count as a root element.

Reviewer: ```suggestion
Note that pages **must have a single root element** to allow route transitions between pages. (Also note that HTML comments count as elements.)
```

Reviewer 2: Can we maybe also an inline example of bad usage with comments?
